src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs

```diff
@@ -179,14 +180,14 @@ public void CompressedData(IApplicationBuilder builder)
             });
     }
 
-    [DllImport("kernel32.dll")]
-    static extern uint GetDllDirectory(uint nBufferLength, [Out] StringBuilder lpBuffer);
+    [LibraryImport("kernel32.dll")]
+    private static partial uint GetDllDirectory(uint nBufferLength, [Out, MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.U2)] char[] lpBuffer);
 
     private async Task DllDirectory(HttpContext context)
     {
-        var builder = new StringBuilder(1024);
-        GetDllDirectory(1024, builder);
-        await context.Response.WriteAsync(builder.ToString());
+        var buffer = new char[1024];
+        GetDllDirectory(1024, buffer);
+        await context.Response.WriteAsync(buffer.ToString());
```

Comment:
This does not look right. `ToString()` on array returns the array type name. So this will always return `System.Char[]` and the value returned from the PInvoke will be ignored.

Reply:
Oh shoot... Seems the result here doesn't matter as the tests didn't fail. That is unfortunate. I will submit a fix.

Reply:
Oh I see. This API isn't use at all anymore. The PR will remove it.

Reply:
Thanks @jkotas. The PR is https://github.com/dotnet/aspnetcore/pull/41957.